wapitiCore/attack/mod_http_headers.py

```diff
@@ -0,0 +1,78 @@
+import requests
+from wapitiCore.attack.attack import Attack
+from wapitiCore.net.web import Request
+from wapitiCore.language.vulnerability import Additional
+#This module check the recommendations of security parameters in HTTP headers
+class mod_http_headers(Attack):
+
+    name = "http_headers"
+    check_list = ['DENY', 'SAMEORIGIN', 'ALLOW-FROM']
+
+    def is_set(self, request: object, header_name, check_list):
+        if header_name == "X-Frame-Options":
+            if 'X-Frame-Options' not in request.headers:
+                return False
+            else:
+                return any(element in request.headers['X-Frame-Options'] for element in check_list)
+        if header_name == "X-XSS-Protection":
+            if "X-XSS-Protection" not in request.headers:
+                return False
+            else:
+                return '1' in request.headers['X-XSS-Protection']
+        if header_name == "X-Content-Type-Options":
+            if "X-Content-Type-Options" not in request.headers:
+                return False
+            else:
+                return "nosniff" in request.headers['X-Content-Type-Options']
+        if header_name =="Strict-Transport-Security":
+            if 'Strict-Transport-Security' not in request.headers:
+                return False
+            else:
+                return "max-age=" in request.headers['Strict-Transport-Security']
+
+    def attack(self):
+        url = self.persister.get_root_url()
+        session = requests.session()
+        req = session.get(url)
```

Comment:
Please use the `crawler` attribute of the attack module in order to use proxy/auth settings specified by the user. Send the request using `self.crawler.send(request)`. It will return a `Page` object that should work with your `is_set` function without modifications

Reply:
Fixed